device/transport/mqtt/devdoc/mqtt_requirements.md

```diff
@@ -341,3 +312,67 @@ The `getTwinReceiver` method creates a `MqttTwinReceiver` object for the twin re
 **SRS_NODE_DEVICE_MQTT_16_064: [** `disableTwin` shall call its callback with no arguments when the `UNSUBACK` packet is received. **]**
 
 **SRS_NODE_DEVICE_MQTT_16_065: [** `disableTwin` shall call its callback with an `Error` if an error is received while unsubscribing. **]**
+
+### message Event
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_16_004: [** If there is a listener for the `message` event, a `message` event shall be emitted for each message received. **]**
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_16_005: [** When a `message` event is emitted, the parameter shall be of type `Message`. **]**
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_16_007: [** When a message is received, the receiver shall populate the generated `Message` object `properties` property with the user properties serialized in the topic. **]**
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_16_008: [** When a message is received, the receiver shall populate the generated `Message` object `messageId` with the value of the property `$.mid` serialized in the topic, if present. **]**
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_16_009: [** When a message is received, the receiver shall populate the generated `Message` object `to` with the value of the property `$.to` serialized in the topic, if present. **]**
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_16_010: [** When a message is received, the receiver shall populate the generated `Message` object `expiryTimeUtc` with the value of the property `$.exp` serialized in the topic, if present. **]**
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_16_011: [** When a message is received, the receiver shall populate the generated `Message` object `correlationId` with the value of the property `$.cid` serialized in the topic, if present. **]**
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_16_012: [** When a message is received, the receiver shall populate the generated `Message` object `userId` with the value of the property `$.uid` serialized in the topic, if present. **]**
+
+### method Event
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_13_003: [** If there is a listener for the `method` event, a `method_<METHOD NAME>` event shall be emitted for each message received. **]**
+
+**SRS_NODE_DEVICE_MQTT_RECEIVER_13_005: [** When a `method_<METHOD NAME>` event is emitted the parameter shall conform to the shape as defined by the interface specified below:
+
+```
+interface StringMap {
+    [key: string]: string;
+}
+
+interface MethodMessage {
+    methods: { methodName: string; };
+    requestId: string;
+    properties: StringMap;
+    body: any;
+    verb: string;
+}
+```
+**]**
+
+### inputMessage event
+
+**SRS_NODE_DEVICE_MQTT_18_033: [** When the `inputMessage` event is subscribed to, if no `moduleId` was specified through the `constructor`, `Mqtt` shall throw an `InvalidOperationError` exception. **]**
+
+**SRS_NODE_DEVICE_MQTT_18_034: [** When the `inputMessage` event is subscribed to, `Mqtt` shall subscribe to the "devices/<deviceId>/modules/<moduleId>/#" topic **]**
+
```

Comment:
I would avoid that - the reason is that event subscription doesn't have a callback so there's no way to feed the error back to the user if the call to subscribe fail. I would rather see a pattern like the enable/disable functions that is being called from the main client (enableModuleInput/disableModuleInput for example)